lib/cartopy/io/shapereader.py

```diff
@@ -186,10 +186,10 @@ class FionaReader:
 
     """
 
-    def __init__(self, filename, bbox=None):
+    def __init__(self, filename, bbox=None, encoding=None):
```

Comment:
```suggestion
    def __init__(self, filename, bbox=None, encoding='utf-8'):
```
Thinking about the comment from @greglucas, there are a lot of different options since the two libraries do slightly different things (pyshp raises a TypeError if encoding=None is passed; fiona attempts to detect the encoding by default while pyshp does not; and both accept different kwargs other than encoding). I believe the simplest and most compatible change would be to simply make utf-8 the default for both. This also does not seem to lose much functionality for fiona; for example I tried reading a gbk encoded shapefile and while fiona did not raise an error, it also did not actually read it correctly without specifying the encoding.

Reply:
I don't think we should restrict users to only using utf-8 by default. If users had previously installed Fiona and were reading in other encodings, they would have been able to use the data, but if we force the encoding now then that would break them. I guess there is a question of whether we should call one of these libraries our defacto standard and go with that and then manipulate keyword arguments to be compatible with that...

I still think passing `**kwargs` through is the most flexible and least amount of work, very similar to what the stock_img pr is doing by forwarding them on to the underlying function: https://github.com/SciTools/cartopy/pull/2230